core/authentication_api.php

```diff
@@ -233,6 +233,19 @@ function auth_credential_page( $p_query_string, $p_user_id = null, $p_username =
 	return $t_auth_flags->getCredentialsPage( $p_query_string );
 }
 
+/**
+ * Gets the page that validates the user credentials based on the standard flow.
+ *
+ * @param string $p_query_string The query string, can be empty.
+ * @param int|null $p_user_id The user id or null for current logged in user.
+ * @param string $p_username The username
+ * @return string The authenticator page with query string.
+ */
+function auth_authenticator_page( $p_user_id = null, $p_username = '' ) {
```

Comment:
The term `authenticator` reflects a specific implementation.  We should probably consider `extended_auth`.  It can work with all kinds of 2FA or any other approach (e.g. security questions), etc.

I would expect such page to be visible only after user enters their correct credentials via the standard password page.  If user is using a custom credentials page, then this setting wouldn't be applicable.

Reply:
The user is either using credentials_page OR authenticator_page. The former request the password from the user and authenticates it (or does SSO), while the later replaces the standard login.php and  handles the authentication (redirects to login.php if unsuccessful).

Reasoning: I wanted to use the standard mantis flow (username and password aquisition), but was not able to change login.php to support the plugin as a second auth mech, since the authflags got resetted to default before the redirect and changing this seemed to be a much deeper change. I named is `authenticator`, because the page actually does the authentication with the username/password POST-ed to it (see SampleAuth readme in PR)